packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -693,6 +695,168 @@ describe('Context bindings - Injecting dependencies of classes', () => {
     await expect(ctx.get(STORE_KEY)).to.be.rejectedWith('Bad');
   });
 
+  it('injects a config property', () => {
+    class Store {
+      constructor(
+        @config('x') public optionX: number,
+        @config('y') public optionY: string,
+      ) {}
+    }
+
+    ctx.configure('store').to({x: 1, y: 'a'});
+    ctx.bind('store').toClass(Store);
+    const store: Store = ctx.getSync('store');
+    expect(store.optionX).to.eql(1);
+    expect(store.optionY).to.eql('a');
+  });
+
+  it('injects a config property with promise value', async () => {
+    class Store {
+      constructor(@config('x') public optionX: number) {}
+    }
+
+    ctx.configure('store').toDynamicValue(() => Promise.resolve({x: 1}));
+    ctx.bind('store').toClass(Store);
+    const store = await ctx.get<Store>('store');
+    expect(store.optionX).to.eql(1);
+  });
+
+  it('injects a config property with a binding provider', async () => {
+    class MyConfigProvider implements Provider<{}> {
+      constructor(@inject('prefix') private prefix: string) {}
+      value() {
+        return {
+          myOption: this.prefix + 'my-option',
+        };
+      }
+    }
+
+    class Store {
+      constructor(@config('myOption') public myOption: string) {}
+    }
+
+    ctx.bind('config').toProvider(MyConfigProvider);
+    ctx.configure('store').toProvider(MyConfigProvider);
+    ctx.bind('prefix').to('hello-');
+    ctx.bind('store').toClass(Store);
+
+    const store = await ctx.get<Store>('store');
+    expect(store.myOption).to.eql('hello-my-option');
+  });
+
+  it('injects a config property with a rejected promise', async () => {
+    class Store {
+      constructor(@config('x') public optionX: number) {}
+    }
+
+    ctx
+      .configure('store')
+      .toDynamicValue(() => Promise.reject(Error('invalid')));
+
+    ctx.bind('store').toClass(Store);
+
+    await expect(ctx.get('store')).to.be.rejectedWith('invalid');
+  });
+
+  it('injects a config property with nested property', () => {
+    class Store {
+      constructor(@config('x.y') public optionXY: string) {}
+    }
+
+    ctx.configure('store').to({x: {y: 'y'}});
+    ctx.bind('store').toClass(Store);
+    const store: Store = ctx.getSync('store');
+    expect(store.optionXY).to.eql('y');
+  });
+
+  it('injects config if the binding key is not present', () => {
```

Comment:
The test name seems to describe a scenario that's different from the actual test code. Is this testing `@config()` invocation with no property accessor? 

The first `@config` argument is called `configPath` and described as `Optional property path of the config.`. Let's use "config property path" instead of "binding key" in the test names.

```suggestion
  it('injects entire config object if there is no config property path present', () => {
```